lib/src/firestore.dart

```diff
@@ -348,6 +367,41 @@ class _FirestoreData {
     setProperty(nativeInstance, key, data);
   }
 
+  void setFieldValue(String key, FieldValue value) {
+    assert(key != null);
+    var data;
+    if (value != null) {
+      switch (value) {
+        case FieldValue.serverTimestamp:
+          data = fieldValueServerTimestamp();
+          break;
+        case FieldValue.delete:
+          data = fieldValueDelete();
+          break;
+        default:
+          throw new ArgumentError.value(
+              value, key, 'Unsupported value type for Firestore.');
+      }
+    }
+    setProperty(nativeInstance, key, data);
+  }
+
+  // only for testing we should never read such value
+  FieldValue getFieldValue(String key) {
```

Comment:
Is it possible to write an integration test and avoid having this method at all?

It seems that we'd just need to set a field value, write to the database, read the document back and verify that our field is either not there (for deleted) or contains a non-empty DateTime?

Reply:
If you read the document data before writing it to Firestore, you will get the FieldValue.serverTimestamp (that's how it works on javascript). After you write to firestore, reading the same field contains DateTime object. As I said in my other comment what would be the value if you call fromMap? But I understant your concert as such fields (delete, serverTimestamp) should be write only!

Reply:
But yes, I will make it private only (will be use in toMap though)